### ISSUE TITLE
stage switch of the default registry to gsoci

### ISSUE THREAD

**Issue by piontec** [giantswarm/roadmap#3410]
[on hold, we already switched CAPx, we are blocked by vintage code to do that for vintage]

The part that blocks us on vintage is related to this:
- https://gigantic.slack.com/archives/C02GDJJ68Q1/p1703841020654699
- https://gigantic.slack.com/archives/C02HLSDH3DZ/p1704717062700159

Unless this explicit dependency on `docker.io` registry is solved by a vintage release, we can't proceed :(
Ping @alex-dabija .

We need to switch the default value in the CCRs to `gsoci.azurecr.io`. This will cause applications' rollout and restart. We have to plan it and do it in stages.

We will switch the default in CCRs, starting with our test installations. Then, we will start switching smaller installations at once and bigger in batches. Only when it's done, we will switch the default in `shared-configs`.

We want to switch this value: https://github.com/giantswarm/shared-configs/blob/main/default/config.yaml#L11-L14

```[tasklist]
### Test installations switch - phase 0
  - [ ] gaia
  - [ ] garfish
  - [ ] gauss
  - [ ] giraffe
  - [ ] glasgow
  - [ ] glippy
  - [ ] golem
  - [ ] gopher
  - [ ] snail
  - [ ] coordinate and split adidas' clusters into batches
  - [ ] coordinate and split vodaphone's clusters into batches
```

```[tasklist]
### Production switch - phase 1
  - [ ] akita
  - [ ] alpaca
  - [ ] anchovy
  - [ ] anteater
  - [ ] antelope
  - [ ] anzu
  - [ ] argali
  - [ ] bumblebee
  - [ ] cuckoo
  - [ ] eagle
  - [ ] fox
  - [ ] gaggle
  - [ ] gazelle
  - [ ] gcapeverde
  - [ ] gerbil
  - [ ] gmc
  - [ ] goat
  - [ ] goose
  - [ ] goshawk
  - [ ] goten
  - [ ] grizzly
  - [ ] leopard
  - [ ] mammoth
  - [ ] potato
  - [ ] puma
  - [ ] valkyrie
  - [ ] velvet
  - [ ] viking
  - [ ] viper
  - [ ] visitor
  - [ ] wallaby
```

```[tasklist]
### Production switch - phase 2
  - [ ] switch the default in `shared-configs`
  - [ ] remove CCR overrides
```

**Comment by alex-dabija** [comment]
@piontec What are the benefits of making this change for Vintage if it depends on having all the workload clusters upgraded for a given installation? The upgrades will probably take a very long time and we are currently pushing for cluster to be migrated to CAPA anyway.

I would like to understand better what happens if we don't change the registry for Vintage clusters. I thought we are currently within the limits with requests to `docker.io`.

**Comment by piontec** [comment]
This change is done for CAPI. As we agreed, it won't be done for vintage, unless something changes a lot. Closing.
